### ISSUE TITLE
does dhewm support the editor? It doesn't work on Linux

### ISSUE THREAD

**Issue by SingingBush** [dhewm/dhewm3#519]
Does dhewm include the editor? I've recently been playing around with it on Windows (steam Doom 3 install) and was hoping to do the same from my Linux machine. I tried opening the editor but the command seems to be unrecognised:

![image](https://github.com/dhewm/dhewm3/assets/870567/27a73a73-9419-4b86-8828-943307759ff8)

It's the same outcome if I specify `+editor` as an arg:

```
./dhewm3 +editor +r_fullscreen 0 +r_mode 7
```

System:
Fedora 38 (Linux Kernel 6.2.15, Gnome 44.1 & Wayland)

**Comment by DanielGibson** [comment]
The editor is windows-only, and so are the other integrated GUI tools (light editor, gui editor, ...).
You could use the DarkRadiant - or try to run dhewm3 in Wine.

**Comment by SingingBush** [comment]
Thanks for the quick response. I'm currently compiling DarkRadiant so will see how that goes. If someone was to make a PR that ports all GUI tools to something like gtk or Qt, is that something you'd consider merging?

**Comment by DanielGibson** [comment]
Yes, I would (as long as it can still be disabled at compiletime to allow dhewm3 builds with fewer dependencies - on Windows the tools can also be disabled).

Possible alternative to Gtk or Qt: https://github.com/ocornut/imgui
There even is some kind of Radiant level editor based on that: https://github.com/xoxor4d/iw3xo-radiant
But it's for Call of Duty 4 (which AFAIK has roots in idTech3, though they probably changed *a lot*, so it would need to be adapted for idtech4/Doom3) and its licensing is quite unclear (I couldn't find any mention of a license in the README, the source files, and neither a License.txt file at the top level of the repo. There is https://github.com/xoxor4d/iw3xo-radiant/blob/develop/readme/licenses.txt but that might only apply to the libs it uses). However, if it turns out to be under a compatible license (and is indeed a usable editor, I never tried it) I could imagine that it could be a good base for a Dear ImGui-based level Editor in Doom3/dhewm3.

**Comment by DanielGibson** [comment]
The iw3xo-radiant developer has confirmed that the whole project is under MIT license: https://github.com/xoxor4d/iw3xo-radiant/issues/26

**Comment by SingingBush** [comment]
Dark Radiant seems like it's the best option for non-windows users to work on Doom 3 maps at the minute. Potentially it could be good to have the editor command start darkradiant if it's on the system path.

I'd not heard of imgui before so have had an interesting time in the last two evenings looking at using it directly from C++ as well as bindings to Java & D. I'm going to spend a bit more time familiarising myself with it.

**Comment by DanielGibson** [comment]
If you want to port iw3xo-radiant to dhewm3, I have a branch with ImGui support now which should be a good start: https://github.com/dhewm/dhewm3/pull/576

**Comment by BraXi** [comment]
> If you want to port iw3xo-radiant to dhewm3, I have a branch with ImGui support now which should be a good start: #576

Seeing _that_ radiant in doom3 would be dream come true.
